### PR TITLE
Add some naming restrictions for Tanagra objects

### PR DESCRIPTION
**Pull request by wchamber**
Add some simple name validation restrictions for Entities, Attributes, and Relationships. We need to refere to these names in APIs and probably don't want to allow anything of any length.

### CODE REVIEW

File: api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import com.google.common.base.Preconditions;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities for checking Tanagra names.
+ *
+ * <p>Some names traverse Tanagra from HTTP requests to SQL generation. For these names, it's
+ * important that they are relatively friendly and simple.
+ */
+final class NameUtils {
```
Comment (calbach) [inline api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java:12]: Should this be applied to variables?

**Review by calbach — APPROVED** [review]
I like it, thanks.

File: api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java (hunk 1 above)
Comment (wchamber) [reply api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java:12]: Maybe. Variables are only user specified and are used directly in SQL, so they're a little differnt. OTOH, having a consistent simple naming schema might be valuable. I could go either way, WDYT?

Comment (calbach) [reply api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java:12]: I can definitely see them needing to diverge for that reason (or else necessitating a conversion layer for Tanagra variable -> SQL variable). I'd keep them semantically separate, e.g. in API documentation - but your call whether you want to just reuse this code, or split out a different utility.

Comment (wchamber) [reply api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java:12]: reverting variable to keep it separate for now.

**Review by calbach — APPROVED** [review]

**Pull request merged**